### PR TITLE
[release-1.13][FLINK-22670][FLIP-150][connector/common] Hybrid source baseline

### PR DESCRIPTION
**Pull request by tweise**
Backport of https://github.com/apache/flink/pull/15924

See https://lists.apache.org/thread.html/rb122520f43637a48e5b92f5d20cf05eedf15f5f7e4e1d7b557173561%40%3Cdev.flink.apache.org%3E

### CODE REVIEW

**Comment by AHeise** [comment]
I'd like to include the documentation as well into this PR. Ideally, e2e test would also be in, but I think this is takes a bit longer.

**Comment by tweise** [comment]
Yes, we can add the documentation commit after https://github.com/apache/flink/pull/16932 os merged.

**Comment by tweise** [comment]
@AHeise documentation commit added, PTAL

File: docs/content/docs/connectors/datastream/hybridsource.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Hybrid Source
+
+`HybridSource` is a source that contains a list of concrete [sources]({{< ref "docs/dev/datastream/sources" >}}).
```
Comment (AHeise) [inline docs/content/docs/connectors/datastream/hybridsource.md:28]: We should add here somewhere that this is only available in 1.13.3 (or whatever the next release is after the backport).

**Review by AHeise — APPROVED** [review]
We should mention the specific version required to use it here to manage expectations. Apart from that LGTM.

**Pull request merged**